src/_pytest/capture.py

```diff
@@ -451,6 +468,10 @@ class MultiCapture:
     _state = None
 
     def __init__(self, out=True, err=True, in_=True, Capture=None):
+        if Capture == OrderedCapture:
+            self.streams = collections.deque()
+            Capture = functools.partial(OrderedCapture, (self.streams))
```

Comment:
This should rather be passed in as a separate `Capture` class, i.e. use `functools.partial` before passing it here.

Reply:
Why do you think that'd be better? The only other place where I can use `fuctools.partial` is in `CaptureFixture` class, but I don't see any added value here.

Reply:
I think the logic should not go into MultiCapture really, but should be handled by the passed in class/callable.

Reply:
I've not checked where/if you can do it in another way really though, but here's an example for having a capture class that resumes on reading from stdin (just as an example really, https://github.com/blueyed/pytest/blob/eda98a194a4b512a1fa5aba8e45f9bbe1f1866a1/src/_pytest/capture.py#L98-L111).

Reply:
I've changed the approach to have the `deque` as class attribute of the `OrderedWriter`.  
Anyway, I'll wait for you to merge #6671 before I'll finish this branch.